File: web/src/main/java/no/nav/modiapersonoversikt/service/unleash/strategier/IsNotProdStrategy.java
package no.nav.modiapersonoversikt.service.unleash.strategier;

import no.finn.unleash.strategy.Strategy;
import no.nav.common.utils.EnvironmentUtils;

import java.util.Map;

public class IsNotProdStrategy implements Strategy {
    @Override
    public String getName() {
        return "isNotProd";
    }

    @Override
    public boolean isEnabled(Map<String, String> map) {
        return !"p".equals(EnvironmentUtils.getOptionalProperty("APP_ENVIRONMENT_NAME").orElse("local"));
    }
}

File: web/src/main/java/no/nav/modiapersonoversikt/service/unleash/strategier/IsNotProdStrategy.kt
package no.nav.modiapersonoversikt.service.unleash.strategier

import io.getunleash.UnleashContext
import io.getunleash.strategy.Strategy
import no.nav.modiapersonoversikt.service.unleash.strategier.StrategyUtils.getApplicationEnvironment

class IsNotProdStrategy : Strategy {
    override fun getName(): String = "isNotProd"

    override fun isEnabled(
        parameters: Map<String, String>,
        context: UnleashContext,
    ): Boolean = getApplicationEnvironment() != "p"
}
